examples/tls-logger/server.cpp

```diff
@@ -42,6 +49,7 @@ class server
                 buffer.data(), 
                 static_cast<uint32_t>(buffer.size()));
 
+            file_lock.lock();
             file.write(reinterpret_cast<const char*>(buffer.data()),
                 buffer.size());
             
```

Comment:
You can unlock the file after you are done writing to it. Maybe just call `file.close()` first.